src/redisraft.c

```diff
@@ -1613,6 +1613,44 @@ static int cmdRaftRandom(RedisModuleCtx *ctx,
     return REDISMODULE_OK;
 }
 
+static bool allowSubscribe(RedisRaftCtx *rr)
+{
+    /* Allow subscription only if we are the leader and the log replay is
+     * completed. Otherwise, if PUBLISH messages are in a multi or lua script,
+     * subscribers can receive it while we are replaying the log entries. */
+    if (rr->state != REDIS_RAFT_UP ||
+        !raft_is_leader(rr->raft) ||
+        raft_get_last_applied_term(rr->raft) != raft_get_current_term(rr->raft)) {
+        return false;
+    }
+
+    return true;
+}
+
+static int cmdRaftRejectSubscribe(RedisModuleCtx *ctx, RedisModuleString **argv,
+                                  int argc)
+{
+    (void) argv;
+    (void) argc;
+
+    RedisRaftCtx *rr = &redis_raft;
+
+    if (checkRaftState(rr, ctx) != RR_OK) {
+        return REDISMODULE_OK;
+    }
+
+    if (!raft_is_leader(rr->raft)) {
+        Node *leader = getLeaderNodeOrReply(rr, ctx);
+        if (leader) {
+            redirectCommand(rr, ctx, leader);
+        }
+        return REDISMODULE_OK;
+    }
+
+    replyClusterDown(ctx);
```

Comment:
maybe worth mentioning in a comment the case here - log replay is not competed yet

Reply:
done